src/open_r1/grpo.py

```diff
@@ -114,7 +116,9 @@ def make_conversation(example):
         }
 
     dataset = dataset.map(make_conversation)
-    dataset = dataset.remove_columns("messages")
+    if "messages" in dataset.column_names:
+        # (Ed) not sure if this is required
+        dataset = dataset.remove_columns("messages")
```

Comment:
Yes, it is: `“message”` is a kind of reserved key in trl. In fact, trl doesn't expect a dataset to have both `messages` and `prompt` keys (it doesn't really make sense anyway).

see https://github.com/huggingface/trl/blob/801582ec240c46ecd487a72fcc8944f268380830/trl/data_utils.py#L77-L85